src/S3/S3ClientTrait.php

```diff
@@ -94,9 +94,14 @@ public function copyAsync(
     /**
      * @see S3ClientInterface::registerStreamWrapper()
      */
-    public function registerStreamWrapper()
+    public function registerStreamWrapper($useV2Existence = false)
     {
-        StreamWrapper::register($this);
+        StreamWrapper::register(
```

Comment:
Should we not add the variable on s3 clients so we don't have to change this call?

Reply:
What do you think about adding a registerStreamWrapperV2) method instead?  I thought about adding it to the S3 client, but this is the only customization that uses object and bucket existence checks.